### PR TITLE
Finish postgres & mysql features (missing sqlite)

### PR DESCRIPTION
**Pull request by GopherJ**

### CODE REVIEW

**Comment by GopherJ** [comment]
@hsluoyz  Hello, could you review it?

File: README.md [hunk 1]
```diff
@@ -1 +1,39 @@
...
+
+```rust
+extern crate casbin;
+extern crate diesel_adapter;
+
+use casbin::{Enforcer, FileAdapter, Model};
+use diesel_adapter::{DieselAdapter, ConnOptions};
+
+let mut m = Model::new();
+m.load_model("examples/rbac_model.conf");
```
Comment (xcaptain) [inline README.md]: You can use `Model::new_from_file("")` to create a model for simplicity

File: README.md [hunk 2]
```diff
@@ -1 +1,39 @@
...
+
+```
+diesel_adapter = { version = "0.1.0", features = ["postgres"] }
+```
+
+
+## Example
+
+```rust
+extern crate casbin;
```
Comment (xcaptain) [inline README.md]: No need to write `extern crate` for rust 2018 edition, just write `use casbin::{Model, Adapter}`

File: src/adapter.rs [hunk 3]
```diff
@@ -0,0 +1,630 @@
...
+        let manager = ConnectionManager::new(conn_opts.get_url());
+        let pool = Pool::builder().build(manager).map_err(Error::PoolError)?;
+
+        cfg_if! {
+            if #[cfg(feature = "postgres")] {
+                pool.get()
+                    .map_err(|err| Box::new(Error::PoolError(err)) as Box<dyn StdError>)
+                    .and_then(|conn| {
+                        sql_query(format!(
+                            r#"
```
Comment (xcaptain) [inline src/adapter.rs]: I prefer not to create table automatically, sometimes a database account doesn't have permission to do create operation.

Comment (hsluoyz) [reply src/adapter.rs]: A common behavior for a Casbin adapter (e.g., all Golang adapters) will try to create a DB named `casbin` if it doesn't exist. If it has no permission, then let it fail.

Comment (GopherJ) [reply src/adapter.rs]: Other adapters like typeorm-adapter (nodejs)  is doing so. 

maybe I should let users choose to create on their own or by this crate.

**Comment by hsluoyz** [comment]
Can we use some more meaningful commit message than `add backup`, `backup 2`?

Also avoid containing merge commits like `Merge pull request #1 from GopherJ/fix/tests `, I think they can be removed by rebase/cherry-pick

**Comment by GopherJ** [comment]
@hsluoyz You are right;) I'll try to modify them later.

**Pull request closed without merging**